### DISCUSSION TITLE
How to calculate the BW, when iperf3 showing Rx BW more than expected and Tx very small

### DISCUSSION THREAD

**Discussion by amd-sathekum** [esnet/iperf#1457] [Q&A]
#2 questions we have, our NIC supports 10G, but we are validating for 100M only,

1. Why are we getting more BW only for Tx and not for Rx. In that case how can we calculate the total BW applied
```
[SUM][RX-C]   0.00-30.00  sec   428 MBytes   120   Mbits/sec  **34811**     sender
[SUM][RX-C]   0.00-30.01  sec   323 MBytes    *90.3 Mbits/sec*                  receiver
```
2. And how bad are those retry values? How to fix those

(Maintainer note: adjusted formatting for legibility)

**Comment by davidBar-On** [comment]
It would help if you could also specify the iperf3 command you run and the report header.  Also, what iperf3 version you are using and what OS (use `iperf3 -v` to get this info).

As the question is about retries, I assume that this is a TCP test and not UDP.

1. In general, the Tx BW is the BW the sender is sending the packets and the Rx BW is the actual BW received.  With TCP the difference may be for several reasons, one of them is that data is still in different network buffers when the the test ended.  It can help if you will show the statistics per second, not just the summary.
2. The retries usually mean that the sending rate is above the network rate.  Try to set different sending rate (using the `-b` option) and see how that effects the retries and BW rate.

**Reply by bmah888** [reply]
Following on to the second point, any place along the path between the sender and receiver that there is a speed difference (say between two router interfaces of difference speeds) creates additional potential for packet drops.

Note that TCP retries are more or less expected as a consequence of TCP trying to figure out the bandwidth along the path, so they're not automatically bad. I can't speak to the number of retries in this particular test though, again it'd be good to see all of the command-line arguments for both client and server to get a better idea of what was being tested.

Bruce.
